### ISSUE TITLE
Add runcircuit to evaluate amplitudes of a set of measurement data

### ISSUE THREAD

**Issue by GTorlai** [GTorlai/PastaQ.jl#212]

**Comment by mtfishman** [comment]
Would the operation involve projecting an MPS onto a series of product states?

**Comment by GTorlai** [comment]
Yes, and I imagine the naive way would be to apply half the circuit to the zero state, and half the circuit (dag) to the product state of interest.

**Comment by GTorlai** [comment]
This functionality now requires contraction graph optimization, which is underway within ITensor
